### PR TITLE
Don't call resolver if domain is already resolved

### PR DESCRIPTION
**Pull request by Bolodya1997**
# Motivation
There is no need to call resolver if domain is already in `tcp://${addr}:${port}` format.

### CODE REVIEW

File: pkg/registry/common/dnsresolve/common.go [hunk 1]
```diff
@@ -37,34 +38,48 @@ type Resolver interface {
...
 	}
 
-	if len(ips) == 0 {
-		return nil, errors.New("resolver.LookupIPAddr return empty result")
-	}
+	return u, nil
+}
 
-	u, err := url.Parse(fmt.Sprintf("tcp://%v:%v", ips[0].IP, records[0].Port))
+var resolvedPattern = regexp.MustCompile("tcp://(?P<addr>[0-9.:]+):(?P<port>[0-9]+)?")
```
Comment (denis-tingaikin) [inline pkg/registry/common/dnsresolve/common.go:73]: Can we just try to parse it as a URL?

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

File: pkg/registry/common/dnsresolve/common.go (hunk 1 above)
Comment (Bolodya1997) [reply pkg/registry/common/dnsresolve/common.go:73]: Nope, any domain is most possible a valid URL. We need a domain in `tcp://ip:port` format.

File: pkg/registry/common/dnsresolve/common.go [hunk 2]
```diff
@@ -37,34 +38,48 @@ type Resolver interface {
 }
 
 func resolveDomain(ctx context.Context, service, domain string, r Resolver) (*url.URL, error) {
-	serviceDomain := fmt.Sprintf("%v.%v", service, domain)
-	_, records, err := r.LookupSRV(ctx, service, "tcp", serviceDomain)
+	var resolved string
+	if isResolved(domain) {
```
Comment (denis-tingaikin) [inline pkg/registry/common/dnsresolve/common.go:42]: please move this check to top-level before call resolve domain.

File: pkg/registry/common/dnsresolve/common.go (hunk 1 above)
Comment (denis-tingaikin) [reply pkg/registry/common/dnsresolve/common.go:73]: Lets then check that part in URL after the scheme is a valid IP and PORT.

You could split Path into two strings entry to parse parts with default parsers for IP and int

**Review by denis-tingaikin — CHANGES_REQUESTED** [review]

**Review by denis-tingaikin — APPROVED** [review]

**Pull request merged**